tools/myst/enc/enc.c

```diff
@@ -154,26 +136,38 @@ static uint64_t _forward_exception_as_signal_to_kernel(
         siginfo.si_signo = SIGSEGV;
         // `si_addr` is always null for #GP.
         siginfo.si_addr = NULL;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
-        return OE_EXCEPTION_CONTINUE_EXECUTION;
+        supported_code = true;
     }
     if (oe_exception_code == OE_EXCEPTION_X87_FLOAT_POINT)
     {
         // ATTN: Consider implementing accurate si-code for
         // OE_EXCEPTION_X87_FLOAT_POINT
         siginfo.si_code = FPE_FLTINV;
         siginfo.si_signo = SIGFPE;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
-        return OE_EXCEPTION_CONTINUE_EXECUTION;
+        supported_code = true;
     }
     if (oe_exception_code == OE_EXCEPTION_DIVIDE_BY_ZERO)
     {
         siginfo.si_code = FPE_INTDIV;
         siginfo.si_signo = SIGFPE;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
+        supported_code = true;
+    }
+
+    if (supported_code)
+    {
+        uint64_t rsp;
+
+        memset(oe_context, 0, sizeof(oe_context_t));
+
+        asm volatile("mov %%rsp, %0" : "=r"(rsp));
```

Comment:
What stack is being used by this function? The Mystikos stack or the OE stack?

After this assembly line, can rsp register change due to the C code that follows? Is it a problem if oe_context->rsp is not based on the most recent rsp register?

Reply:
By default, the exception handler will continue to use the stack left at when the AEX occurs. So as long as it occurs during the Mystikos execution, the stack will be Mystikos stack.

The rsp/rbp is fixed per function, only expected to be changed at the beginning and end of the function. Continue with the current rsp allows the mystikos signal handler to use parameters saved on the stack (e.g., `mcontext`). If we save the parameters as a global variable, they could be overwritten by nested exceptions.

Reply:
The current value of RSP here could be
- OE stack
- Alternate Exception Stack registered with OE

There is a potential problem in the latter case.

myst_handle_host_signal is scheduled to be executed using this RSP value.
Upon return from this function, OE will clear the state and then `oe_continue_execution` to `myst_handle_host_signal` using the exception stack.

Now, if an exception is triggered early enough in `myst_handle_host_signal`, then OE will reuse the  alternate exception stack. which could potentially corrupt the stack being used by the incomplete `myst_handle_host_signal` call.

Speaking of which, is `myst_handle_host_signal` reentrant?

Reply:
I will go ahead and approve the PR. The case I mentioned can be handled when we look at alternate stack.